autogluon/utils/tabular/ml/trainer/abstract_trainer.py

```diff
@@ -727,42 +744,39 @@ def compress(self, X=None, y=None, models=None):
 
         models_compressed = {}
         model_levels = defaultdict(dd_list)
+        ignore_models = []
+        ignore_stack_names = [REFIT_FULL_NAME]
+        for stack_name in ignore_stack_names:
+            ignore_models += self.get_model_names(stack_name)  # get_model_names returns [] if stack_name does not exist
         for model_name in models:
             model = self.load_model(model_name)
-            if isinstance(model, WeightedEnsembleModel):
+            if isinstance(model, WeightedEnsembleModel) or model_name in ignore_models:
                 continue
             model_level = self.get_model_level(model_name)
-            model_levels['compressed'][model_level] += [model_name]
-            model_compressed = model.convert_to_compressed_template()
+            model_levels[REFIT_FULL_NAME][model_level] += [model_name]
+            model_compressed = model.convert_to_refitfull_template()
             models_compressed[model_name] = model_compressed
-        levels = sorted(model_levels['compressed'].keys())
+        levels = sorted(model_levels[REFIT_FULL_NAME].keys())
         models_trained_full = []
         for level in levels:
-            models_level = model_levels['compressed'][level]
+            models_level = model_levels[REFIT_FULL_NAME][level]
             models_level = [models_compressed[model_name] for model_name in models_level]
-            models_trained = self.stack_new_level_core(X=X, y=y, models=models_level, level=level, stack_name='compressed', hyperparameter_tune=False, feature_prune=False, kfolds=0, n_repeats=1)
+            models_trained = self.stack_new_level_core(X=X, y=y, models=models_level, level=level, stack_name=REFIT_FULL_NAME, hyperparameter_tune=False, feature_prune=False, kfolds=0, n_repeats=1)
             models_trained_full += models_trained
         return models_trained_full
 
-    def distill(self, X=None, y=None):
-        if X is None:
-            X = self.load_X_train()
-        if y is None:
-            y = self.load_y_train()
-
-        model_best = self.load_model(self.model_best)
-        if self.problem_type == MULTICLASS:
-            raise NotImplementedError
-        if not self.bagged_mode:
-            raise NotImplementedError
-        models_distill = get_preset_models_distillation(path=self.path, problem_type=self.problem_type, objective_func=self.objective_func, stopping_metric=self.stopping_metric, num_classes=self.num_classes, hyperparameters=self.hyperparameters)
-        y_distill = pd.Series(model_best.oof_pred_proba)
-
-        # TODO: Do stratified for binary/multiclass, folds are not aligned!
-        models_trained = self.stack_new_level_core(X=X, y=y_distill, models=models_distill, level=0, stack_name='distilled', hyperparameter_tune=False, feature_prune=False)
-        self.compress(X=X, y=y_distill, models=models_trained)
-
-        self.save()
+    def best_single_model(self, stack_name, stack_level):
```

Comment:
Looking at the original intent of this PR (Although it seems the scope has expanded), this is the only function that is necessary for the original intended PR, correct?

Reply:
If you mean getting "compressed" models, then yes this is the only function needed for that.
The rest of the PR contains other internally-accessible functionality that will be useful for the closely related task of distillation and other problem_types such as SOFTCLASS.